data/json.go

```diff
@@ -152,6 +154,15 @@ type jsonConnector struct {
 
 func init() {
 	register("json", func(connStr string, l *common.Logger) (DataConnector, error) {
+
+		if _, err := os.Stat(filepath.Dir(connStr)); os.IsNotExist(err) {
+			os.MkdirAll(filepath.Dir(connStr), 0777)
```

Comment:
os.MkdirAll returns an error - please dont throw it away, instead please handle it